File: Backend/Data/Models/Game.cs

namespace Backend.Data.Models;

public class Game
{
    public int Id { get; set; }
    public DateTime StartTime { get; set; } = DateTime.Now;
    public DateTime? EndTime { get; set; }
    public int SequenceId { get; set; }
    public virtual required Sequence Sequence { get; set; }
    public virtual required ICollection<Player> Players { get; set; }
}

File: Frontend/app/models/Game.ts
import { Sequence } from "./Sequence";
import { GamePlayer } from "./GamePlayer";
import { GameAction } from "./GameAction";

export interface Game {
  id: number;
  name?: string;
  createdAt: Date;
  updatedAt?: Date;
  startTime?: Date;
  endTime?: Date;
  gameToken?: string;
  sequenceId: number;
  sequence: Sequence;
  players: GamePlayer[];
  actions: GameAction[];
}
